storage/appinsights/appinsights.go

```diff
@@ -60,48 +73,66 @@ func (Storage) Type() string {
 // Store takes a list of Checker results and sends them to the configured
 // Application Insights instance.
 func (c Storage) Store(results []types.Result) error {
+	c.TelemetryClient = appinsights.NewTelemetryClientFromConfig(c.TelemetryConfig)
+	for k, v := range c.Tags {
+		c.TelemetryClient.Context().CommonProperties[k] = v
+	}
 	for _, result := range results {
 		c.send(result)
 	}
 	return c.close()
 }
 
 // Close will submit all queued telemetry to the configured Application Insights
-// service.
+// service.  If either RetryInterval or MaxRetries are <= 0, Close() should proceed
+// without a retry and a maximum timeout of 2 seconds.
 // Ref: https://github.com/microsoft/ApplicationInsights-Go#shutdown
 func (c Storage) close() error {
+	if c.RetryInterval <= 0 || c.MaxRetries <= 0 {
+		select {
+		case <-c.TelemetryClient.Channel().Close():
+			return nil
+		case <-time.After(2 * time.Second):
+			return fmt.Errorf("Failed to submit telemetry during close")
+		}
+	}
 	select {
-	case <-c.TelemetryClient.Channel().Close(10 * time.Second):
-		// Ten second timeout for retries.
+	case <-c.TelemetryClient.Channel().Close(c.RetryInterval * time.Second):
 		return nil
-	case <-time.After(30 * time.Second):
+	case <-time.After((c.MaxRetries + 1) * c.RetryInterval * time.Second):
 		return fmt.Errorf("Failed to submit telemetry after retries")
 	}
 }
 
 // Send a result
 // Multiple test result measurements will be added to the telemetry item's
-// common properties
+// customMeasurements field
 func (c Storage) send(conclude types.Result) {
 	total := int64(0)
-	message := "Passed"
+
+	message := "Up"
 	if conclude.Degraded {
 		message = "Degraded"
 	} else if conclude.Down {
 		message = "Down"
 	}
-	message = fmt.Sprintf("%s ", conclude.Notice)
-	mean := time.Duration(total / int64(len(conclude.Times)))
-	availability := appinsights.NewAvailabilityTelemetry(conclude.Title, mean, conclude.Healthy)
+	if conclude.Notice != "" {
+		message = fmt.Sprintf("%s - %s ", message, conclude.Notice)
+	}
+
+	availability := appinsights.NewAvailabilityTelemetry(conclude.Title, 0, conclude.Healthy)
 	availability.RunLocation = c.TestLocation
 	availability.Message = message
 	availability.Id = uuid.New().String()
-	availability.GetProperties()["ThresholdRTT"] = conclude.ThresholdRTT.String()
 	for i := 0; i < len(conclude.Times); i++ {
 		k := strconv.Itoa(i)
 		total += int64(conclude.Times[i].RTT)
 		availability.GetMeasurements()[k] = float64(conclude.Times[i].RTT)
 	}
```

Comment:
Instead of calculating Mean (or avg, max, min), you can use Result.ComputeStats. You can use this value higher up in NewAvailabilityTelemetry as well (second parameter, based on previous commit)